src/java/org/apache/cassandra/cql3/restrictions/StatementRestrictions.java

```diff
@@ -63,6 +69,14 @@ public final class StatementRestrictions
             "Executing this query despite the performance unpredictability with ALLOW FILTERING has been disabled " +
             "by the allow_filtering_enabled property in cassandra.yaml";
 
+    public static final String ANN_REQUIRES_INDEX_MESSAGE = "ANN ordering by vector requires the column to be indexed";
+
+    public static final String VECTOR_INDEXES_ANN_ONLY_MESSAGE = "Vector indexes only support ANN queries";
+
+    public static final String ANN_ONLY_SUPPORTED_ON_VECTOR_MESSAGE = "ANN ordering is only supported on float vector indexes";
+
+    public static final String ANN_REQUIRES_INDEXED_FILTERING_MESSAGE = "ANN ordered queries can only be filtered with indexed columns";
```

Comment:
Might want to grab the improvements from ds vsearch dc3cc020

Reply:
Unfortunately, the `StatementRestrictions` implementation is very different on ds vsearch but I have picked up the improved error message.